pkg/addons/addons.go

```diff
@@ -145,14 +145,18 @@ func enableOrDisableAddon(cc *config.ClusterConfig, name string, val string) err
 
 	// to match both ingress and ingress-dns addons
 	if strings.HasPrefix(name, "ingress") && enable {
-		if driver.IsKIC(cc.Driver) && runtime.GOOS != "linux" {
-			exit.Message(reason.Usage, `Due to networking limitations of driver {{.driver_name}} on {{.os_name}}, {{.addon_name}} addon is not supported.
+		if driver.IsKIC(cc.Driver) {
+			if runtime.GOOS == "windows" {
```

Comment:
Would the same approach would work for macOS as well? It seems like it should.

Reply:
I don't have access to a macOS machine. Could you please try the following and see if the addon is working?
1. Pull the latest from the master branch and build for macos.
2. `kubectl apply -f kubectl apply -f https://projectcontour.io/quickstart/contour.yaml`
3. `kubectl apply -f https://projectcontour.io/examples/kuard.yaml`
4. `minikube tunnel`
5. Try accessing kuard. If you are able to access kuard using Contour Ingress then we can enable the ingress addon for macos, otherwise not.

Reply:
@blueelvis I just tried it&mdash;this is the output (repeated every 5s):
```
Status:	
	machine: minikube
	pid: 15490
	route: 10.96.0.0/12 -> 172.16.63.33
	minikube: Running
	services: [envoy]
    errors: 
		minikube: no errors
		router: no errors
		loadbalancer emulator: no errors
```

and this is http://172.16.63.33:

![Screen Shot 2020-12-16 at 8 05 45 AM](https://user-images.githubusercontent.com/54780737/102353490-1f141500-3f77-11eb-8ead-7e1d32cea3af.png)

Reply:
But that was from `master`.  Did you mean to checkout your branch `blueelvis:task/enable-ingress-docker-windows`?

Reply:
@bl-ue - Did you test this on minikube with Docker driver on MacOS? Or did minikube create a dedicated VM for you? And yes, I meant taking the latest master branch.

Reply:
@blueelvis Yes, it did create a vm because I started minikube with `--vm=true`.  Would you like me to test using Docker, i.e. without `--vm=true`?

Reply:
Oops! I wasn't even using my freshly built `minikube` 😄   Let me retest with Docker and minikube `master` v1.16.0-beta.0.

Reply:
So this time, `minikube tunnel` doesn't produce any output after I enter my password:
```
$ minikube tunnel
❗  The service envoy requires privileged ports to be exposed: [80 443]
🔑  sudo permission will be asked for it.
🏃  Starting tunnel for service envoy.
Password:
```
- http://127.0.0.1:80 presents me with the same screen as in the screenshot above (with different IDs of course).
- https://127.0.0.1:443 shows `This site can’t be reached ... The connection was reset.`

Reply:
FWIW, it did not work with me on macOS with the Docker driver. `kubectl get services` never showed an IP, and localhost:443/80, while listened on, didn't go to the service in question (`curl: (56) Recv failure: Connection reset by peer`).

I think we're close, but I guess not quite there yet.

Reply:
Thanks both of you for testing this!

Reply:
You're so welcome @blueelvis!!!!